composio/local_tools/ragtool/actions/rag_add_request.py

```diff
@@ -4,25 +4,29 @@
 
 
 class RagToolAddRequest(BaseModel):
-    content: str = Field(..., description="Content to add to the knowledge base", json_schema_extra={"file_readable": True})
+    content: str = Field(
+        ...,
+        description="Content to add to the knowledge base",
+        json_schema_extra={"file_readable": True},
+    )
 
 
 class RagToolAddResponse(BaseModel):
     status: str = Field(..., description="Status of the addition to the knowledge base")
 
 
-class AddContentToRagTool(Action):
+class AddContentToRagTool(Action[RagToolAddRequest, RagToolAddResponse]):
     """
     Tool for adding content to the knowledge base
     """
 
     _display_name = "Add Content to Rag Tool"
-    _request_schema = RagToolAddRequest
-    _response_schema = RagToolAddResponse
+    _request_schema: type[RagToolAddRequest] = RagToolAddRequest
+    _response_schema: type[RagToolAddResponse] = RagToolAddResponse
     _tags = ["Knowledge Base"]
     _tool_name = "ragtool"
 
-    def execute(self, request: RagToolAddRequest, authorisation_data: dict = None):
+    def execute(self, request: RagToolAddRequest, authorisation_data: dict) -> dict:
```

Comment:
```suggestion
    def execute(self, request_data: RagToolAddRequest, authorisation_data: dict) -> dict:
```

https://github.com/ComposioHQ/composio/blob/42debbddf3cb2f0e7d77f3f6eaff2f9ea9495a2e/composio/core/local/action.py#L81